src/main/java/com/cmlteam/serv/HttpHandlerFilesList.java

```diff
@@ -0,0 +1,121 @@
+        httpExchange.sendResponseHeaders(200, 0);
+        OutputStream os = httpExchange.getResponseBody();
+        String param = findRequestParam(httpExchange.getRequestURI().toString());
+        File nextFolder;
+        switch (param) {
+            case "toplevel":
+                showList(files, os);
+                break;
+            case "back":
+                if (referers.size() < 2) {
+                    referers.clear();
+                    showList(files, os);
+                } else {
+                    referers.pop();
+                    nextFolder = new File (referers.peek());
+                    showList(nextFolder.listFiles(), os);
+                }
+                break;
+            default:
+                referers.push(param);
+                nextFolder = new File(param);
+                showList(nextFolder.listFiles(), os);
+        }
+        os.flush();
+        os.close();
+    }
+
+    private void showList(File[] folder, OutputStream os) throws IOException{
+        for (String s: HEADER) {
+            os.write(s.getBytes(StandardCharsets.UTF_8));
+        }
+        for (File file: folder) {
+            if (file.isDirectory()) {
+                writeClickableFolderName(os, file);
+            }
+            else if (file.isFile()) {
+                writeNonClickableFileName(os, file);
+            } else {
+                System.out.println(file.getName() + " is not supported");
+            }
+        }
+        for (String str: FOOTER) {
+            os.write(str.getBytes(StandardCharsets.UTF_8));
+        }
+    }
+
+    private void writeNonClickableFileName (OutputStream os, File file) throws IOException{
+        String name = file.getName();
+        double dsize = file.length()/1000;
+        String size = String.format("%5.2f", dsize) + "KB";
+        String[] content = {"<tr>",
+                                "<td>" + name + "</td>",
+                                "<td>" + size + "</td>",
+                            "</tr>"
+        };
+        for (String s: content) {
+            os.write(s.getBytes(StandardCharsets.UTF_8));
+        }
+    }
+
+    private void writeClickableFolderName (OutputStream os, File file) throws IOException{
+        String name = file.getName();
+        String convertedName = file.getAbsolutePath().replace("\\", "$" );
+        String[] content = {"<tr>",
+                                "<td><a href='/listing?name=" + convertedName + "'>" + name + "</a></td>",
+                                "<td>---</td>",
+                            "</tr>"
+        };
+        for (String s: content) {
+            os.write(s.getBytes(StandardCharsets.UTF_8));
+        }
+    }
+
+    private String findRequestParam (String url) {
+        StringTokenizer st = new StringTokenizer(url, "=");
+        if (st.countTokens() >=2) {
+            st.nextToken();
+            return st.nextToken().replace("$", "\\").replace("%20", " ");
+        }
```

Comment:
When i created manually the folder `aaa=bbb` i was not able to open it, though I should.